src/Lecture3.hs

```diff
@@ -122,10 +136,14 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) :: Reward -> Reward -> Reward
+  Reward { rewardGold = g1, rewardSpecial = s1 }
+    <> Reward { rewardGold = g2, rewardSpecial = s2 }
+      = Reward { rewardGold = g1 <> g2, rewardSpecial = s1 || s2 }
 
 instance Monoid Reward where
-
+  mempty :: Reward
+  mempty  = Reward { rewardGold = mempty :: Gold, rewardSpecial = False }
```

Comment:
Due to type inference, Haskell knows that `mempty` here has type `Gold` so you don't need to specify it explicitly
```suggestion
  mempty  = Reward { rewardGold = mempty, rewardSpecial = False }
```